server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java

```diff
@@ -119,7 +120,10 @@ public static Query refExistsQuery(Reference ref, Context context, boolean count
         String field = ref.column().fqn();
         FieldType fieldType = context.queryShardContext().getMapperService().getLuceneFieldType(field);
         DataType<?> valueType = ref.valueType();
-        boolean canUseFieldsExist = ref.hasDocValues() || (fieldType != null && !fieldType.omitNorms());
+        boolean canUseFieldsExist = fieldType != null && (
+            (fieldType.indexOptions() != IndexOptions.NONE && fieldType.omitNorms() == false) || // indexes norms
+            (fieldType.vectorDimension() != 0) || // indexes vectors
+            (fieldType.docValuesType() != DocValuesType.NONE)); // indexes doc-values
```

Comment:
Referenced https://github.com/apache/lucene/blob/57dd5a4bda89d417174f3e672a657b1176c6fbeb/lucene/core/src/java/org/apache/lucene/search/FieldExistsQuery.java#L111

Reply:
This flag does not work as intended since the `FieldType` used when creating a column and when indexing a value are different and the one needed for `IsNullPredicate` is the latter. Also, to my understanding obtaining the `FieldType` is a deterministic process, hence `Reference.luceneFieldType()`, `Reference.isFieldNameIndexed` are introduced.